pkg/operator/observe_config.go

```diff
@@ -92,12 +90,43 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+	observedConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&observedConfig)
+	currentStatus := operatorConfig.Status.DeepCopy()
+
+	var errs []error
+	for _, observer := range c.observers {
+		var currErrs []error
+		observedConfig, currErrs = observer(c.kubeClient, &rest.Config{}, observedConfig)
+		errs = append(errs, currErrs...)
 	}
 
-	glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
-	operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
+	if len(errs) > 0 {
+		var messages []string
+		for _, currentError := range errs {
+			messages = append(messages, currentError.Error())
+		}
+		v1alpha1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, v1alpha1.OperatorCondition{
+			Type:    v1alpha1.OperatorStatusTypeFailing,
+			Status:  v1alpha1.ConditionTrue,
+			Reason:  configObservationError,
+			Message: strings.Join(messages, "\n"),
+		})
+	} else {
+		condition := v1alpha1helpers.FindOperatorCondition(operatorConfig.Status.Conditions, v1alpha1.OperatorStatusTypeFailing)
+		if condition != nil && condition.Status == v1alpha1.ConditionTrue && condition.Reason == configObservationError {
+			condition.Status = v1alpha1.ConditionFalse
+			condition.Reason = configObservationSuccess
+			condition.Message = "Configuration observed successfully"
+		}
+	}
+	if reflect.DeepEqual(currentConfig, observedConfig) && reflect.DeepEqual(currentStatus, operatorConfig.Status) {
+		return nil
+	}
+	if !reflect.DeepEqual(currentConfig, observedConfig) {
```

Comment:
ok, this code is weird.  Just check the spec update, do it.  Then check the status update (from the result of the first update!) and do that.